### PR TITLE
Added flows and atoms for starting/stopping volume rebalance

### PR DESCRIPTION
**Pull request by shtripat**
The status of the rebalance would be taken care as part of
cluster data sync.

Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by r0h4n** [comment]
@shtripat Please resolve conflicts

**Comment by r0h4n** [comment]
Please resolve merge conflicts

**Comment by shtripat** [comment]
@r0h4n rabsed

**Comment by r0h4n** [comment]
Please resolve merge conflicts

**Comment by r0h4n** [comment]
Any updates?

**Comment by shtripat** [comment]
@r0h4n This would need change to use gdeploy for starting rebalance. Will take this up next.

**Comment by shtripat** [comment]
Pending verification.
@nnDarshan plz review.

File: tendrl/gluster_integration/objects/definition/gluster.yaml [hunk 1]
```diff
@@ -300,6 +300,29 @@ namespace.gluster:
...
+            mandatory:
+              - Volume.volname
+              - Volume.vol_id
+            optional:
+              - Volume.fix_layout
+              - Volume.force
+          name: start_rebalance
+          run: gluster.objects.Volume.atoms.StartRebalance
+          type: Start
+          uuid: 242f6190-9b37-11e6-950d-a24fc0d96599
```
Comment (nnDarshan) [inline tendrl/gluster_integration/objects/definition/gluster.yaml]: help missing in above 2 atoms ?

Comment (shtripat) [inline tendrl/gluster_integration/objects/definition/gluster.yaml]: oh. will add theme. thanks.

**Comment by shtripat** [comment]
I can add one for stop as well if required. Do we have gdeploy wrapper for the same already?

**Comment by nnDarshan** [comment]
@shtripat we have to use the same method rebalance_volume() with action as "stop"

**Comment by shtripat** [comment]
@nnDarshan added stop rebalance as well here. Plz review..

File: tendrl/gluster_integration/objects/volume/atoms/rebalance_not_running/__init__.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        )
+        try:
+            rebal_status = NS.etcd_orm.client.read(
+                'clusters/%s/Volumes/%s/rebal_status' % (
+                    NS.tendrl_context.integration_id,
+                    self.parameters['Volume.vol_id']
+                )
+            ).value
+            if rebal_status is not None:
+                if rebal_status == "not applicable" or\
```
Comment (nnDarshan) [inline tendrl/gluster_integration/objects/volume/atoms/rebalance_not_running/__init__.py]: not_started ?

File: tendrl/gluster_integration/objects/volume/atoms/rebalance_running/__init__.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                'clusters/%s/Volumes/%s/rebal_status' % (
+                    NS.tendrl_context.integration_id,
+                    self.parameters['Volume.vol_id']
+                )
+            ).value
+            if rebal_status is not None:
+                if rebal_status == "not applicable" or\
+                    rebal_status == "completed":
+                    return False
+                if rebal_status == "in progress":
```
Comment (nnDarshan) [inline tendrl/gluster_integration/objects/volume/atoms/rebalance_running/__init__.py]: can we have just this check, if status == in progress return true, else false ?

File: tendrl/gluster_integration/objects/volume/atoms/stop_rebalance/__init__.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    obj = Volume
+    def __init__(self, *args, **kwargs):
+        super(StopRebalance, self).__init__(*args, **kwargs)
+
+    def run(self):
+        if NS.gdeploy_plugin.rebalance_volume(
+                self.parameters.get('Volume.volname'),
+                "stop",
+                force=self.parameters['Volume.force'],
+                fix_layout=self.parameters['Volume.fix_layout']
```
Comment (nnDarshan) [inline tendrl/gluster_integration/objects/volume/atoms/stop_rebalance/__init__.py]: I feel not needed

File: tendrl/gluster_integration/objects/definition/gluster.yaml [hunk 5]
```diff
@@ -300,6 +300,56 @@ namespace.gluster:
...
+          uuid: 242f6190-9b37-11e6-950d-a24fc0d96579
+        StopRebalance:
+          enabled: true
+          help: stop rebalance
+          inputs:
+            mandatory:
+              - Volume.volname
+              - Volume.vol_id
+            optional:
+              - Volume.fix_layout
```
Comment (shtripat) [inline tendrl/gluster_integration/objects/definition/gluster.yaml]: Ok. Will remove the same.

File: tendrl/gluster_integration/objects/definition/gluster.yaml [hunk 6]
```diff
@@ -358,6 +408,46 @@ namespace.gluster:
...
+          atoms:
+            - gluster.objects.Volume.atoms.StopRebalance
+          help: "Stop Volume Rebalance"
+          enabled: true
+          inputs:
+            mandatory:
+              - Volume.volname
+              - Volume.vol_id
+            optional:
+              - Volume.fix_layout
```
Comment (shtripat) [inline tendrl/gluster_integration/objects/definition/gluster.yaml]: Will remove this.

File: tendrl/gluster_integration/objects/volume/atoms/rebalance_not_running/__init__.py (hunk 2 above)
Comment (shtripat) [inline tendrl/gluster_integration/objects/volume/atoms/rebalance_not_running/__init__.py]: Do we have not_started state as well? I remember it was "not applicable" if not started at all and "completed" if earlier one is complete. If "not_started" as well there, can be added in conditions here.

File: tendrl/gluster_integration/objects/volume/atoms/rebalance_running/__init__.py (hunk 3 above)
Comment (shtripat) [inline tendrl/gluster_integration/objects/volume/atoms/rebalance_running/__init__.py]: That a good suggestion I feel. will do that.

File: tendrl/gluster_integration/objects/volume/atoms/stop_rebalance/__init__.py (hunk 4 above)
Comment (shtripat) [inline tendrl/gluster_integration/objects/volume/atoms/stop_rebalance/__init__.py]: Will remove this.

**Comment by shtripat** [comment]
@r0h4n verifying the flows now.

**Comment by shtripat** [comment]
Verified the PR and works as expected.

**Pull request merged**